angular/src/app/utils/questionnaireBuilder.ts

```diff
@@ -0,0 +1,2990 @@
+          class: 'optionsWrapper',
+          children: [
+            {
+              tag: 'p',
+              class: 'editFieldLabel inline boldText redText',
+              html: 'Required',
+            },
+          ],
+        },
+        { tag: 'p', class: 'headingText', html: this.heading },
+        { tag: 'p', class: 'questionText', html: this.label },
+        { tag: 'p', class: instruction_class, html: this.instructions },
+        {
+          tag: 'div',
+          class: 'wrapper alignLeft',
+          children: [
+            {
+              tag: 'div',
+              class: 'editModeMap locationMapContainer',
+              children: [{ tag: 'div', class: 'mapSelectionIcon' }],
+            },
+            { tag: 'p', class: 'locationQuestionDetails inline', html: 'Lat' },
+            { tag: 'input', class: 'locationLatLon' },
+            { tag: 'p', class: 'locationQuestionDetails inline', html: 'Lon' },
+            { tag: 'input', class: 'locationLatLon' },
+            { tag: 'p', class: 'editFieldLabel', html: 'Address' },
+            { tag: 'textarea', class: 'locationAddress' },
+          ],
+        },
+        {
+          tag: 'div',
+          class: 'wrapper',
+          children: [
+            {
+              tag: 'div',
+              class: 'buttonDecline disabled hidden positionLeft',
+              html: 'Prefer not to answer',
+            },
+          ],
+        },
+      ],
+    });
+
+    if (!this.heading)
+      $(this.view_question_DOM).find('p.headingText').addClass('hidden');
+
+    if (parent_question && answer_index > -1) {
+      this.view_question_DOM = DOM.new({
+        tag: 'div',
+        class: 'questionWrapper subQuestion indent',
+        children: [
+          { tag: 'p', class: 'questionText smallText', html: this.label },
+          {
+            tag: 'div',
+            class: 'wrapper centerText',
+            children: [
+              { tag: 'textarea', class: 'shortAnswer textFieldPlaceHolder' },
+            ],
+          },
+        ],
+      });
+
+      let view_field = $(
+        $(parent_question.view_question_DOM).find('.questionOption')[
+          answer_index
+        ]
+      );
+
+      $(this.view_question_DOM).insertAfter($(view_field));
+
+      parent_question.options[answer_index].sub_question = this.metadata;
+    }
+
+    if (this.decline) this.addDeclineOption();
+    // this.decline
+    this.initMap();
+  }
+
+  addDeclineOption() {
+    let question = this;
+    let parent_template = question.parent_template;
+
+    if (!question.decline) {
+      question.decline = { label: 'Prefer not to answer' };
+    } else {
+      if (!question.decline.label)
+        question.decline = { label: 'Prefer not to answer' };
+    }
+
+    $($(question.view_question_DOM).find('.buttonDecline')).html(
+      question.decline.label
+    );
+    $($(question.view_question_DOM).find('.buttonDecline')).removeClass(
+      'hidden'
+    );
+  }
+
+  initMap() {

```

Comment:
is this needed?

Reply:
I see some other functions related to creating a questionnaire. can they be removed? or a follow jira issue for that?